x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/analysis/Verifier.java

```diff
@@ -125,7 +126,21 @@ else if (p.resolved()) {
                 var aggs = agg.aggregates();
                 int size = aggs.size() - groupings.size();
                 aggs.subList(0, size).forEach(unresolvedExpressions);
-            } else {
+            }
+            // similar approach for Lookup
+            else if (p instanceof Lookup lookup) {
+                // first check the table
+                var tableName = lookup.tableName();
+                if (tableName instanceof Unresolvable u) {
+                    failures.add(fail(tableName, u.unresolvedMessage()));
+                }
+                // only after that check the match fields
+                else {
+                    lookup.matchFields().forEach(unresolvedExpressions);
+                }
+            }
```

Comment:
Added this in to avoid telling the fields are not found if the table does not exist (which is evident).